kolibri/core/content/utils/content_request.py

```diff
@@ -0,0 +1,525 @@
+                logger.info(
+                    "Imported content metadata for {} out of {} nodes".format(
+                        processed_count, total_count
+                    )
+                )
+        else:
+            logger.warning(
+                "Failed to import content metadata for {}".format(contentnode_id)
+            )
+    logger.info("Imported content metadata for {} nodes".format(processed_count))
+
+
+def _process_metadata_import(incomplete_downloads_without_metadata):
+    """
+    Processes metadata import for a queryset already filtered to those without metadata
+    :param incomplete_downloads_without_metadata: a ContentDownloadRequest queryset
+    :type incomplete_downloads_without_metadata: django.db.models.QuerySet
+    """
+    peer = _get_preferred_network_location(version_filter=">=0.16.0")
+    if not peer:
+        # can't import metadata without peers having minimum version 0.16.0
+        logger.info("No acceptable peer network device for importing content metadata")
+        return
+
+    # during processing, if there's a critical failure in making requests to the peer, this will
+    # capture those errors, and obviously the raise exceptions will interrupt processing
+    with capture_connection_state(peer):
+        with NetworkClient.build_from_network_location(peer) as client:
+            # test connection
+            client.connect()
+            _import_metadata(
+                client,
+                incomplete_downloads_without_metadata.values_list(
+                    "contentnode_id", flat=True
+                ),
+            )
+
+
+def _process_content_requests(incomplete_downloads):
+    """
+    Processes content requests, both for downloading and removing content
+    """
+    incomplete_downloads_with_metadata = incomplete_downloads.filter(has_metadata=True)
+
+    # obtain the incomplete removals, that do not have an associated download
+    incomplete_removals = (
+        ContentRemovalRequest.objects.annotate(
+            has_download=Exists(
+                ContentDownloadRequest.objects.filter(
+                    contentnode_id=OuterRef("contentnode_id")
+                )
+            )
+        )
+        .filter(
+            status__in=INCOMPLETE_STATUSES,
+            has_download=False,
+        )
+        .order_by("requested_at")
+    )
+    incomplete_sync_removals = incomplete_removals.filter(
+        reason=ContentRequestReason.SyncInitiated
+    )
+    incomplete_user_removals = incomplete_removals.filter(
+        reason=ContentRequestReason.UserInitiated
+    )
+    complete_user_downloads = ContentDownloadRequest.objects.filter(
+        status=ContentRequestStatus.Completed, reason=ContentRequestReason.UserInitiated
+    )
+
+    # loop while we have pending downloads
+    while incomplete_downloads_with_metadata.exists():
+        # grab the next request that will fit within current free space
+        free_space = get_free_space()
```

Comment:
`get_free_space` defaults to the KOLIBRI_HOME dir which by default will be on the same drive as the content directory, but is configurable to be elsewhere.

Should be:
```
free_space = get_free_space(OPTIONS["Paths"]["CONTENT_DIR"])
```